src/pickup.c

```diff
@@ -2443,15 +2452,30 @@ exchange_objects_with_mon(struct monst *mtmp, boolean taking)
                     /* extra delay for removing a cloak */
                     m_delay += 2;
                 }
-                pline("%s %s %s %s.", Monnam(mtmp),
-                      m_delay > 1 ? "begins removing" : "removes",
-                      mhis(mtmp), xname(otmp));
+                if ((unwornmask & W_SADDLE) != 0) {
+                    You("remove %s from %s.", the(xname(otmp)),
+                        x_monnam(mtmp, ARTICLE_THE, (char *) 0,
+                                 SUPPRESS_SADDLE, FALSE));
+                    /* unstrapping a saddle takes additional time */
+                    time_taken += rn2(3);
+                }
```

Comment:
If this is now supported, why not just delete the special case in the caller that gives a y/n prompt for removing the saddle? I don't see the point of having two ways to do it.